### PR TITLE
added primitive match return function to CDMS reader

### PR DESCRIPTION
**Pull request by alovett-COAPS**
Fulfils requirement from SDAP-470- adding in functionality to cdms_reader.py to return matches in an alternate, simplified format

- returns primary to secondary match information in format "matches[m][0-8]", where m is the match and indices 0-8 represent all associated primary and secondary data for that match
- for users who attempt to use the "assemble_matches" function directly, this function can be used instead to obtain match information in a different format.

### CODE REVIEW

File: tools/cdms/cdms_reader.py [hunk 1]
```diff
@@ -1,250 +1,316 @@
...
+            sec_lat = []
+            sec_time = []
+            sec_speed = []
+            sec_dir = []
+
+        sec_lon.append(float(secondary.lon[int(y)].values))
+        sec_lat.append(float(secondary.lat[int(y)].values))
+        sec_time.append(float(secondary.time[int(y)].values))
+        sec_speed.append(float(secondary.wind_speed[int(y)].values))
+        sec_dir.append(float(secondary.wind_to_direction[int(y)].values))
```
Comment (ngachung) [inline tools/cdms/cdms_reader.py:153]: @alovett-COAPS This change to the reader appears to be specific just for the primary = sst and secondary = wind use case. This reader should be generic to support any variable(s). In the use case where secondary is an in situ source there can be also be multiple variables returned.

**Review by ngachung — CHANGES_REQUESTED** [review]

**Review by alovett-COAPS — COMMENTED** [review]
Updated function to support more use cases than just the wind-sst matchup, updated documentation to reflect changes.

File: tools/cdms/README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ Imported packages:
 * csv
 * collections
 * logging
-    
+* xarray
```
Comment (ngachung) [inline tools/cdms/README.md:17]: @alovett-COAPS It appears xarray isn't used at all in cdms_reader.py so I think we should remove it as a dependency.

**Review by ngachung — APPROVED** [review]
Looks good. Thanks Amanda!

**Review by ngachung — APPROVED** [review]

**Comment by ngachung** [comment]
Changes are good, but CHANGELOG.md needs to be synced with develop and update added to UNRELEASED section.

**Pull request merged**